python/ray/_private/metrics.py

```diff
@@ -0,0 +1,67 @@
+import functools
+import time
+from ray.util.metrics import Counter, Histogram
+from typing import Dict, Callable
+
+
+class FunctionMetricsCollector:
+    """A helper class that implements the context manager and decorator
+    for recording function/block execution.
+    """
+    HISTOGRAMS: Dict[str, Histogram] = dict()
```

Comment:
Do we need this class variable? Why can't we just store 3 metrics to each instance? This seems to have issues like, if you have two same methods with the same name, it will conflict? Or is it intentional to avoid having same name?

Reply:
that's correct; the counter/histogram name will be a deterministic function of this name.